### ISSUE TITLE
Support @ErrorHandler for RequestHandler

### ISSUE THREAD

**Issue by dlydiard** [L2jLiga/fastify-decorators#26]
I tried adding a ErrorHandler in my handler but got some typescript errors. Are there any plans to support the decorator for handlers? I really like the concept, as it would produce really clean code and would give the handle method even better encapsulation and focus.

**Comment by L2jLiga** [comment]
Do you mean using ErrorHandlers withing RequestHandler like in code follow or ?.. 

```typescript
@GET('/')
export default class GetHandler extends RequestHandler {
  async handle() {
    // some logic here
  }

  @ErrorHandler(CustomError)
  handleCustomError(request, reply, done) {
    // some logic here
  }
}
```

**Comment by dlydiard** [comment]
yes, is this supposed to work? if so, i'll try to repeat the error i received tomorrow.

**Comment by L2jLiga** [comment]
> this supposed to work?

not yet, `@ErrorHandler` available only in controllers atm

**Comment by L2jLiga** [comment]
I think it's possible to add it to `RequestHandler` when fastify 3.2.0 published, see https://github.com/fastify/fastify/pull/2427.

**Comment by L2jLiga** [comment]
Fastify 3.2.0 published 🎉 
Here we go: https://github.com/L2jLiga/fastify-decorators/blob/v3/lib/docs/Request-Handlers.md#error-handling

**Comment by dlydiard** [comment]
works great, thanks!

**Comment by L2jLiga** [comment]
Released as [v3.2.0](https://github.com/L2jLiga/fastify-decorators/releases/tag/v3.2.0)
